pkg/account/deployer/client.go

```diff
@@ -96,6 +112,108 @@ func (c *accountManagementClient) getManagementZones(ctx context.Context) ([]Man
 	return envResources.ManagementZoneResources, nil
 }
 
+func (c *accountManagementClient) upsertBoundary(ctx context.Context, boundaryId string, boundary Boundary) (remoteId, error) {
+	if boundaryId == "" {
+		logr.FromContextOrDiscard(ctx).V(1).Info("Trying to get boundary with name " + boundary.Name)
+		bnd, err := c.getBoundaryByName(ctx, boundary.Name)
+		if err != nil {
+			var rnfErr *ResourceNotFoundError
+			if !errors.As(err, &rnfErr) {
+				return "", err
+			}
+
+			logr.FromContextOrDiscard(ctx).V(1).Info("No boundary with name " + boundary.Name + " found. Creating a new one")
+			return c.createBoundary(ctx, boundary)
+		}
+		boundaryId = bnd.Uuid
+	}
+
+	return c.updateBoundary(ctx, boundaryId, boundary)
+}
+
+func (c *accountManagementClient) updateBoundary(ctx context.Context, boundaryId string, boundary Boundary) (string, error) {
+	logr.FromContextOrDiscard(ctx).V(1).Info("Trying to update existing boundary with name " + boundary.Name + " and UUID " + boundaryId)
+	_, resp, err := c.client.PolicyManagementAPI.PutPolicyBoundary(ctx, boundaryId, c.accountInfo.AccountUUID).PolicyBoundaryDto(boundary).Execute()
+	defer closeResponseBody(resp)
+
+	// handle a 404 here if need be as handleClientResponseError discards it!
+	if is404(resp) {
+		return "", ResourceNotFoundError{Identifier: boundaryId}
+	}
+
+	if err = handleClientResponseError(resp, err, "unable to update boundary with name: "+boundary.Name); err != nil {
+		return "", err
+	}
+	return boundaryId, nil
+}
+
+func (c *accountManagementClient) createBoundary(ctx context.Context, boundary Boundary) (string, error) {
+	createdBoundary, resp, err := c.client.PolicyManagementAPI.PostPolicyBoundary(ctx, c.accountInfo.AccountUUID).PolicyBoundaryDto(boundary).Execute()
+	defer closeResponseBody(resp)
+	if err = handleClientResponseError(resp, err, "unable to create boundary with name: "+boundary.Name); err != nil {
+		return "", err
+	}
+
+	if createdBoundary == nil {
+		return "", errors.New("the received data are empty")
+	}
+
+	return createdBoundary.Uuid, nil
+}
+
+func (c *accountManagementClient) getBoundaryByName(ctx context.Context, name string) (*accountmanagement.PolicyBoundaryOverview, error) {
+	boundaries, err := c.getBoundaries(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var foundBoundary *accountmanagement.PolicyBoundaryOverview
+	for _, b := range boundaries {
+		if b.Name == name {
+			if foundBoundary != nil {
+				return nil, fmt.Errorf("found multiple boundaries with name '%s'", name)
+			}
+			foundBoundary = &b
+		}
+	}
+	if foundBoundary == nil {
+		return nil, &ResourceNotFoundError{Identifier: name}
+	}
+
+	return foundBoundary, nil
+}
+
+func (c *accountManagementClient) getBoundaries(ctx context.Context) ([]accountmanagement.PolicyBoundaryOverview, error) {
+	boundaries := make([]accountmanagement.PolicyBoundaryOverview, 0)
```

Comment:
Nit: if you initialize without len/cap, don't use make

`boundaries := []accountmanagement.PolicyBoundaryOverview{}`

Fun fact, here you would not need it at all (just use a var declaration without initalization). `append` already initializes it correctly if it's empty. However, to keep the semantics of the return value (not nil, maybe thats required), we can still use the above statement without the `make` call.